### PR TITLE
Architecture change

### PR DESCRIPTION
**Pull request by ThudPoland**
Extended .gitignore.
Added world classes and interfaces.
Renamed src to server.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,9 @@ cmake_install.cmake
 install_manifest.txt
 compile_commands.json
 CTestTestfile.cmake
+*.a
+*.o
+.vscode/*
+unit_test
+UberFighters-3000-Server
+UberFighters-3000
```
Comment (Kicer86) [inline .gitignore]: Build in build dir, not in sources. You will avoid tons of problems

File: server/Basics/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+set(Basics_sources
+	IWorld.h
+	World.cpp
+	World.h)
+
+add_library(Basics OBJECT ${Basics_sources})
```
Comment (Kicer86) [inline server/Basics/CMakeLists.txt]: why not add all sources to server executable?

File: server/main.cpp [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+#include <SFML/Graphics.hpp>
+#include "Basics/World.h"
+#include <memory>
+
+int main()
+{
+    auto world = std::make_unique<World>();
```
Comment (Kicer86) [inline server/main.cpp]: World world;

Comment (Xavrax) [reply server/main.cpp]: > World world;

Why not as a ptr?

Comment (Kicer86) [reply server/main.cpp]: 1. using stack is faster
2. World world is shorter
3. World world is cleaner
4. using heap in this case has no pros

**Comment by Kicer86** [comment]
> Extended .gitignore.
> Added world classes and interfaces.
> Renamed src to server.

three things. three commits ;)

**Comment by Kicer86** [comment]
Plus: https://github.com/erlang/otp/wiki/writing-good-commit-messages (afair Linus himself said the same)

File: server/main.cpp (hunk 3 above)
Comment (Kicer86) [reply server/main.cpp]: 5. allocating memory on heap may fail

File: server/Basics/CMakeLists.txt (hunk 2 above)
Comment (Xavrax) [reply server/Basics/CMakeLists.txt]: 'file(GLOB helloworld_SRC
    "*.h"
    "*.cpp"
)'
Wouldn't be better?

File: server/main.cpp (hunk 3 above)
Comment (ThudPoland) [reply server/main.cpp]: Will fix. :)

File: server/Basics/CMakeLists.txt (hunk 2 above)
Comment (ThudPoland) [reply server/Basics/CMakeLists.txt]: Why object approach is bad? Classical makefile created 1 object file per cpp file?

Comment (Kicer86) [reply server/Basics/CMakeLists.txt]: @Xavrax https://cmake.org/cmake/help/v3.14/command/file.html#glob
@ThudPoland I do not understand

**Pull request merged**